nodejs/device/src/index.ts

```diff
@@ -359,24 +395,34 @@ class DeviceApp {
         },
       })
     );
+    logger.info(`Sending message with serviceAckId: ${serviceAckId}`);
     this.client
       .sendEvent(message)
       .catch(() => {
-        console.warn(
-          "Error sending message with serviceAckId: " + serviceAckId
+        logger.error(
+          `Error sending message with serviceAckId: ${serviceAckId}`
         );
       })
       .finally(() => {
         this.serviceAckWaitList.add(serviceAckId);
+        if (
+          this.serviceAckWaitList.size > settings.sendMessageAllowedFailureCount
+        ) {
+          return this.stopDeviceApp(
```

Comment:
>return this.stopDeviceApp( [](start = 10, length = 26)

Also, as a philosophical point, I know why you're exiting the app here instead of throwing an exception, but what's the "_right_" way to return a failure from an interval?  In other words, what if we wanted this code to somehow "return" the exception (whatever that means) instead of just exiting the app?  What is the most node-ish way to do this?

Reply:
EventEmitter?

Reply:
Promises could work. Something like this:
```
function failableInterval() {
    return new Promise((resolve, reject) => {
        const timer = setInterval(() => {
            // do stuff
            if (/*failed*/) {
                clearInterval(timer);
                reject(/*err*/);
            }
        }, timeout);
    }
}
```